### ISSUE TITLE
practical steps for launch

### ISSUE THREAD

**Issue by maelle** [cynkra/cynkrablog#16]
@pat-s we're not there yet just now but I thought it'd be useful to know. :pray: 

Eventually how will we launch cynkra.github.io/cynkrablog to cynkra.com/blog?

**Comment by pat-s** [comment]
I'd create a cloudfront instance which will serve the site from an S3 bucket (similar to what we do for cynkra.com). So the publishing workflow would need to sync the output files to an s3 bucket of ours.

We do this for enhanced performance and security control (compared to gh-pages).

**Comment by pat-s** [comment]
You only need the part which syncs the output to S3 (https://github.com/cynkra/cynkraweb/blob/main/.github/workflows/s3-push.yaml#L365-L377). Everything else is done behind the scenes on AWS.

Just let me know once everything is ready.

**Comment by pat-s** [comment]
You can try and I can help and do the cloudfront stuff later on.

**Comment by maelle** [comment]
realizing we probably also need some sort of preview :eyes:

**Comment by krlmlr** [comment]
We can also implement a pull workflow (cynkraweb pulls from cynkrablog). Let's see what works best.

**Comment by christophsax** [comment]
Do we still need something to get this running? @maelle, please let me know if I can help to move this forward.

**Comment by pat-s** [comment]
We need at least a Cloudfront instance and DNS mapping to get this going. I was waiting on some action from your side, see https://github.com/cynkra/cynkrablog/issues/16#issuecomment-1725935826. 

We'd also need an S3 bucket to store the source files. 

Workflows: most of what cynkraweb uses could just be c/p.

**Comment by christophsax** [comment]
@pat-s many people with many ideas in the thread, it's not obvious to me what to do. Cloudfront and DNS mapping could be set up on your side, @pat-s?

@maelle is in charge for this, so she should say what to do next.

**Comment by maelle** [comment]
I wanted to get the HTML checks working before that but maybe it's not needed (I planned to work on it next Monday). Sorry about the delay!

**Comment by christophsax** [comment]
No worries at all. There are many things to figure out. I just would like to ensure it's not hanging at a particular point for some reason. If we have blockers, I am happy to discuss and help to unblock them. Thanks a lot!

**Comment by pat-s** [comment]
I am also just here to help, not to complicate things 🙂

**Comment by maelle** [comment]
what's needed in https://github.com/cynkra/cynkrablog/pull/23? can you adapt it @pat-s or do you need more info?

I'll work on the HTML validation separately. The idea would be to add it later, that is to say, it shouldn't be a blocker.

**Comment by pat-s** [comment]
An S3 bucket `cynkraweb-blog` now exists which is used by a Cloudfront instance to serve content. Once #23 is pushed to main, files will be pushed to that bucket and the site blog.cynkra.com will be live (!). 

If you want to not publish it for now while still being able to merge into `main`, consider disabling the S3 sync step.
